node_test.go

```diff
@@ -2863,57 +2890,87 @@ var nodeEncodeDecodeTests = []struct {
 	},
 }}
 
-func (s *S) TestNodeEncodeDecode(c *C) {
+func TestNodeEncodeDecode(t *testing.T) {
 	for i, item := range nodeEncodeDecodeTests {
-		c.Logf("Encode/Decode test value #%d: %#v", i, item.value)
+		t.Logf("Encode/Decode test value #%d: %#v", i, item.value)
 
 		var v interface{}
 		err := item.node.Decode(&v)
-		c.Assert(err, IsNil)
-		c.Assert(v, DeepEquals, item.value)
+		if err != nil {
+			t.Fatalf("Decode got error: %v", err)
+		}
+		if !reflect.DeepEqual(v, item.value) {
+			t.Fatalf("Decoded value mismatch.\nGot: %+v\nWant: %+v", v, item.value)
+		}
 
 		var n yaml.Node
 		err = n.Encode(item.value)
-		c.Assert(err, IsNil)
-		c.Assert(n, DeepEquals, item.node)
+		if err != nil {
+			t.Fatalf("Encode got error: %v", err)
+		}
+		if !reflect.DeepEqual(n, item.node) {
+			t.Fatalf("Encoded node mismatch.\nGot: %+v\nWant: %+v", n, item.node)
+		}
 	}
 }
 
-func (s *S) TestNodeZeroEncodeDecode(c *C) {
+func TestNodeZeroEncodeDecode(t *testing.T) {
 	// Zero node value behaves as nil when encoding...
 	var n yaml.Node
 	data, err := yaml.Marshal(&n)
-	c.Assert(err, IsNil)
-	c.Assert(string(data), Equals, "null\n")
+	if err != nil {
+		t.Fatalf("Marshal got error: %v", err)
+	}
+	if string(data) != "null\n" {
+		t.Fatalf("Encoded YAML mismatch.\nGot: %q\nWant: %q", string(data), "null\n")
+	}
 
 	// ... and decoding.
 	var v *struct{} = &struct{}{}
-	c.Assert(n.Decode(&v), IsNil)
-	c.Assert(v, IsNil)
+	err = n.Decode(&v)
+	if err != nil {
+		t.Fatalf("Decode got error: %v", err)
+	}
+	if v != nil {
+		t.Fatalf("Decoded value mismatch.\nGot: %+v\nWant: nil", v)
+	}
 
 	// ... and even when looking for its tag.
-	c.Assert(n.ShortTag(), Equals, "!!null")
+	if n.ShortTag() != "!!null" {
+		t.Fatalf("ShortTag mismatch. Got: %q, Want: %q", n.ShortTag(), "!!null")
+	}
```

Comment:
Here also I would recommend using a variable for "!!null", as it is used for detection and reporting